tests/performance/event_generator/src/event_generator/main.py

```diff
@@ -0,0 +1,193 @@
+
+    Raises:
+        ValueError: If any required arguments are missing or invalid.
+
+    Ensures that the 'config' argument is provided and that it points to a valid file.
+    """
+    if not args.config:
+        raise ValueError("Configuration file must be provided.")
+    if not os.path.isfile(args.config):
+        raise ValueError("Invalid configuration file.")
+
+
+def create_event_generator(file_config: dict[str, Any]) -> EventGenerator:
+    """Create and return an EventGenerator instance based on the provided file configuration.
+
+    Args:
+        file_config (dict): Configuration settings specific to the module.
+
+    Returns:
+        EventGenerator: An instance of the appropriate EventGenerator subclass.
+
+    Raises:
+        ValueError: If any required parameters are missing or if the module is unsupported.
+    """
+    module = file_config.get('module')
+    if not module:
+        raise ValueError("Missing 'module' parameter in configuration.")
+
+    # Mapping of module names to their classes and parameters
+    module_mapping = {
+        'logcollector': {
+            'class': LogEventGenerator,
+            'required_params': ['path', 'operations', 'rate'],
+            'optional_params': ['max_file_size', 'template_path', 'max_retries'],
+        },
+        'syscheck': {
+            'class': SyscheckEventGenerator,
+            'required_params': ['path', 'rate', 'num_files', 'num_modifications'],
+            'optional_params': ['max_retries'],
+        },
+    }
+
+    if module not in module_mapping:
+        raise ValueError(f"Unsupported module specified in the configuration: {module}")
+
+    module_info = module_mapping[module]
+    generator_class = module_info['class']
+    required_params = module_info['required_params']
+    optional_params = module_info.get('optional_params', [])
+
+    # Validate required parameters
+    missing_params = [param for param in required_params if param not in file_config]
+    if missing_params:
+        raise ValueError(f"Missing required config parameters for {module}: {', '.join(missing_params)}")
+
+    # Prepare arguments for the generator class
+    init_args = {param: file_config[param] for param in required_params}
+    # Include optional parameters if they are provided
+    init_args.update({param: file_config[param] for param in optional_params if param in file_config})
+
+    # Handle module-specific logic
+    if module == 'syscheck':
+        num_files = init_args['num_files']
+        num_modifications = init_args['num_modifications']
+        # Calculate total operations: create + (modify * num_modifications) + delete
+        operations = num_files + (num_files * num_modifications) + num_files
+        init_args['operations'] = operations
+    elif module == 'logcollector':
+        # For logcollector, operations are provided directly
+        pass
+
+    return generator_class(**init_args)
```

Comment:
This code is difficult to maintain. I suggest:


- Try to create an EventGeneratorFactory class instead. Something like this:
```python
class EventGeneratorFactory:
    def create_event_generator(self, module_name, config):
        if module_name == "logcollector":
            return LogEventGenerator(config)
        elif module_name == "syscheck":
            return SyscheckEventGenerator(config)
        else:
            raise ValueError(f"Unsupported module: {module_name}") 
```

Then move that logic to the event generator module.

- Dont validate the parameters in the function, leave the constructor to make that job
- Don't calculate the operations. This logic is difficult to understand. The user interface should always be capable of providing the number of operations per second.
- Remove the unnecessary logcollector if conditional

Reply:
Fixed here: https://github.com/wazuh/wazuh-qa/pull/5697/commits/376d046f24b879eaa8c5dc0b196dc7a352fc3722